common/lib/xmodule/xmodule/modulestore/tests/factories.py

```diff
@@ -320,47 +323,106 @@ def check_number_of_calls(object_with_method, method_name, maximum_calls, minimu
     return check_sum_of_calls(object_with_method, [method_name], maximum_calls, minimum_calls)
 
 
+class StackTraceCounter(object):
+    def __init__(self, include_arguments=True):
+        self.include_arguments = include_arguments
+
+        if self.include_arguments:
+            self.stacks = defaultdict(lambda: defaultdict(int))
+        else:
+            self.stacks = defaultdict(int)
+
+    def capture_stack(self, args, kwargs):
+        stack = traceback.extract_stack()[:-2]
+
+        if self.include_arguments:
+            safe_args = []
+            for arg in args:
+                try:
+                    safe_args.append(repr(arg))
+                except Exception as e:
+                    safe_args.append('<un-repr-able value: {}'.format(e))
+
+            safe_kwargs = {}
+            for key, kwarg in kwargs.items():
+                try:
+                    safe_kwargs[key] = repr(kwarg)
+                except Exception as e:
+                    safe_kwargs[key] = '<un-repr-able value: {}'.format(e)
+
+            self.stacks[tuple(stack)][tuple(safe_args), tuple(safe_kwargs.items())] += 1
+        else:
+            self.stacks[tuple(stack)] += 1
+
+    @property
+    def total_calls(self):
+        return sum(self.stack_calls(stack) for stack in self.stacks)
+
+    def stack_calls(self, stack):
+        if self.include_arguments:
+            return sum(self.stacks[stack].values())
+        else:
+            return self.stacks[stack]
+
+    @property
+    def stack_list(self):
+        return sorted(self.stacks.keys(), key=self.stack_calls, reverse=True)
+
+
+def capture_call(fn, include_arguments=True):
+    stacks = StackTraceCounter(include_arguments)
+
+    @functools.wraps(fn)
+    def capture(*args, **kwargs):
+        stacks.capture_stack(args, kwargs)
+        return fn(*args, **kwargs)
+
+    capture.__stack_counter = stacks
+
+    return capture
+
```

Comment:
Yeah - guess the difference is that the `call_stack_manager` is counting partial stacks and no arguments.